R/test-DataClass.R

```diff
@@ -0,0 +1,188 @@
+#' deaths_new, deaths_total and that region level have the correct types.
+#' @param data The data to check
+#' @param level character_array the level of the data to check
+#' @param localised logical to check localised data or not, defaults to
+#' TRUE.
+expect_processed_cols <- function(data, level = "1", localised = TRUE) {
+  testthat::expect_s3_class(data[["date"]], "Date")
+  testthat::expect_type(data[["cases_new"]], "double")
+  testthat::expect_type(data[["cases_total"]], "double")
+  testthat::expect_type(data[["deaths_new"]], "double")
+  testthat::expect_type(data[["deaths_total"]], "double")
+  if (!localised) {
+    level_region_str <- paste0("level_", level, "_region")
+    testthat::expect_type(data[[level_region_str]], "character")
+  }
+}
+
+#' Test that cleaned columns contain data/
+#' @description Checks that cleaned columns cases, deaths, recovered and test
+#' (new and total) are not entirely composed of NAs.
+#' @param cntry_obj The DataClass object (R6Class) object to perform checks on.
+#' @importFrom purrr map walk
+#' @importFrom dplyr filter
+#' @importFrom rlang !!
+expect_columns_contain_data <- function(cntry_obj) {
+  cols_present <- function(col) {
+    if (length(cntry_obj$source_data_cols[grep(
+      col, tolower(cntry_obj$source_data_cols)
+    )]) > 0) {
+      return(paste0(col, c("_new", "_total")))
+    } else {
+      return(NULL)
+    }
+  }
+  cols <- c("cases", "deaths", "recovered", "test")
+  cols2check <- map(cols, cols_present)
+  cols2check <- unlist(cols2check)
+  walk(
+    cols2check,
+    ~ {
+      testthat::test_that(
+        paste0(
+          cntry_obj$data_name, "column '", .x, "' is not just composed of NA"
+        ),
+        {
+          testthat::expect_true(
+            nrow(cntry_obj$data$processed %>% filter(!is.na(!!.x))) > 0
+          )
+        }
+      )
+    }
+  )
+}
+
+#' Test download method works correctly
+#' @description Test data can be downloaded if download = TRUE, or a requested
+#' snapshot file is not found, and store a snap shot at the snapshot_path. If an
+#' existing snapshot file is found then load this data to use in future tests.
+#' @param cntry_obj The DataClass object (R6Class) object to perform checks on.
+#' @param download logical check to download or use a snapshot of the data
+#' @param snapshot_path character_array the path to save the downloaded
+#' snapshot to.
+#' @importFrom purrr map walk
+#' @importFrom dplyr slice_tail
+test_download <- function(cntry_obj, download, snapshot_path) {
+  if (!file.exists(snapshot_path)) {
+    download <- TRUE
+  }
+  if (download) {
+    testthat::test_that(paste0(cntry_obj$data_name, " downloads sucessfully"), {
+      cntry_obj$download()
+      walk(cntry_obj$data$raw, function(data) {
+        testthat::expect_s3_class(data, "data.frame")
+        testthat::expect_true(nrow(data) > 0)
+        testthat::expect_true(ncol(data) >= 2)
+      })
+    })
+    cntry_obj$data$raw <- map(cntry_obj$data$raw,
```

Comment:
I thought we only tested on a snapshot when reading from disk. This will always trim to the last few data points which I think we don't want?

Reply:
I just left this as it was in the previous tests

Reply:
ignore me found the error with it

Reply:
fair enough. Can we change here as I think this was a mistake/bug/woops.

Reply:
This is the same as in the tests for regular download which also uses slice_tail to get the last 250 rows